src/Discord.Net.WebSocket/Entities/Interaction/SocketBaseCommand/SocketCommandBase.cs

```diff
@@ -0,0 +1,164 @@
+using Discord.Rest;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using DataModel = Discord.API.ApplicationCommandInteractionData;
+using Model = Discord.API.Interaction;
+
+namespace Discord.WebSocket
+{
+    public class SocketCommandBase : SocketInteraction
+    {
+        /// <summary>
+        ///     The data associated with this interaction.
+        /// </summary>
+        new internal SocketCommandBaseData Data { get; }
+
+        internal SocketCommandBase(DiscordSocketClient client, Model model, ISocketMessageChannel channel)
+            : base(client, model.Id, channel)
+        {
+            var dataModel = model.Data.IsSpecified ?
+                (DataModel)model.Data.Value
+                : null;
+
+            ulong? guildId = null;
+            if (this.Channel is SocketGuildChannel guildChannel)
+                guildId = guildChannel.Guild.Id;
+
+            Data = SocketCommandBaseData.Create(client, dataModel, model.Id, guildId);
+        }
+
+        new internal static SocketInteraction Create(DiscordSocketClient client, Model model, ISocketMessageChannel channel)
+        {
+            var entity = new SocketSlashCommand(client, model, channel);
```

Comment:
Shouldn't this be implemented in each inherited class separately? this way SocketSlashCommand will be created for slash, message and user commands, won't it?

Reply:
oops. fixed. thanks.